design/bgp-bfd.md

```diff
@@ -0,0 +1,157 @@
+# BFD enablement in BGP
+
+## Summary
+
+The purpose of this enhancement is to use enable [BFD](https://en.wikipedia.org/wiki/Bidirectional_Forwarding_Detection) to be used in conjunction with BGP.
+
+BFD will be available only in the upcoming [FRR based implementation](0001-frr.md).
```

Comment:
Not sure I understand the comment. Here we are saying BFD would be enabled in the frr based implementation only and not in the "legacy", in-tree one.

Reply:
I guess @cgoncalves means that it could also be supported by some 3rd party speaker implementation?  That's true, but I don't think we need to make mention of it.  While using a 3rd party speaker with the metallb controller is possible, it's not explicitly supported.

Reply:
Right. And a more general comment is we should strive to create implementation-agnostic APIs. I briefly checked some 3rd party SDN solutions with BFD support and the herein proposed API seems to accommodate them as well.

Reply:
That's great.  A comment like "the proposal here has been spot checked against the configuration of other BFD implementations (including X Y Z), and this design seems to fit well with them, as well."

Reply:
I am not totally comfortable with that, so I'd skip that part.